account_invoice_change_currency/tests/test_account_invoice_change_currency.py

```diff
@@ -0,0 +1,334 @@
+                "reconcile": True,
+            }
+        )
+        self.account_account_line = self.env["account.account"].create(
+            {
+                "name": "acc inv line test",
+                "code": "X2021",
+                "user_type_id": self.account_type2.id,
+                "reconcile": True,
+            }
+        )
+        self.product_1 = self.env["product.product"].create({"name": "Product 1"})
+        self.product_2 = self.env["product.product"].create({"name": "Product 2"})
+        self.analytic_account = self.env["account.analytic.account"].create(
+            {"name": "test account"}
+        )
+        self.tax_account = self.env["account.account"].search(
+            [
+                (
+                    "user_type_id",
+                    "=",
+                    self.env.ref("account.data_account_type_current_assets").id,
+                )
+            ],
+            limit=1,
+        )
+
+    def create_journal_type(self, type_string=None):
+        if not type_string:
+            type_string = "purchase"
+        sequence = self.env["ir.sequence"].create(
+            {
+                "name": "Journal",
+                "prefix": type_string,
+                "padding": 6,
+                "company_id": self.env.ref("base.main_company").id,
+            }
+        )
+        account_journal_type = self.env["account.journal"].create(
+            {
+                "name": "Journal",
+                "code": type_string,
+                "type": type_string,
+                "sequence": sequence.id,
+            }
+        )
+        return account_journal_type
+
+    def create_simple_invoice(
+        self, date=False, context=None, inv_type=None, journal=None
+    ):
+        if not context:
+            context = {}
+        if not journal:
+            journal = self.create_journal_type(type_string="purchase")
+        invoice_lines = [
+            (
+                0,
+                0,
+                {
+                    "product_id": self.product_1.id,
+                    "quantity": 5.0,
+                    "price_unit": 142.0,
+                    "name": "Product that cost 142",
+                    "account_id": self.account_account_line.id,
+                },
+            ),
+            (
+                0,
+                0,
+                {
+                    "product_id": self.product_2.id,
+                    "quantity": 4.0,
+                    "price_unit": 213.0,
+                    "name": "Product that cost 213",
+                    "account_id": self.account_account_line.id,
+                },
+            ),
+        ]
+        invoice = (
+            self.env["account.move"]
+            .with_user(self.manager)
+            .with_context(**context)
+            .create(
+                {
+                    "partner_id": 1,
+                    "move_type": inv_type or "in_invoice",
+                    "journal_id": journal.id,
```

Comment:
Why this change is necessary?

FYI if you pass `default_move_type` as context, default journal will be computed automatically.

Reply:
Changes are done!